### ISSUE TITLE
Hypergraph head

### ISSUE THREAD

**Issue by maxitg** [maxitg/SetReplace#494]
### The problem
Is your feature request related to a problem? Please describe. Ex. I'm always frustrated when [...]

**Comment by maxitg** [comment]
The box should be some kind of an interpretation box (ideally with a purple frame like `Graph` if possible). Otherwise, if you copy it and use it as an input, it does not work.

And yes, the format should be `Hyperedge[{vertices...}]` because, as you say, we need the second argument for other functionality.

**Comment by daneelsan** [comment]
Yes, that was really a quick implementation. As you said, the idea would be to `TimeConstrain` it and show the Graph-like box if it takes too much time.

**Comment by maxitg** [comment]
We can probably estimate how long it would take and constrain based on the number of vertices/hyperedges.

**Comment by taliesinb** [comment]
I'd like to suggest the following syntax for when it comes time to implement symmetries on hyperedges:

* DirectedHyperedge[{1, 2, 3}, {4, 5, 6}]
* UndirectedHyperedge[{1, 2, 3}, {4, 5, 6}] (which is the same as UndirectedHyperedge[{4, 5, 6}, {1, 2, 3}])
* ColoredHyperedge9[{1 → 1, 2 → 1, 3 → 2, 4 → 2}] (which is ColoredHyperedge[{1, 2, 3, 4} → {1, 2, 2}])

These all evaluate to Hyperedge[vertexlist, symmetrygroup].
The symmetry for colors is fully symmetric *within each color*. 
The symmetry for directed and undirected hyperedges is fully symmetric within the source, and the destination separately.
